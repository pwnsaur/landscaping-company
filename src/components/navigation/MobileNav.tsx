import { MouseEvent } from 'react';
import styled, { keyframes } from 'styled-components';

import NavItems from '@components/navigation/NavItems';

const slideIn = keyframes`
  0% {
    transform: translateY(-100%);
  }
  100% {
    transform: translateY(0);
  }
`;

const slideOut = keyframes`
  0% {
    transform: translateY(0);
    visibility: visible;
  }
  100% {
    transform: translateY(-100%);
    visibility: hidden;
  }
`;

type MobileNavProps = {
  isOpen: boolean;
  isVisible: boolean;
  handleItemClick: (event: MouseEvent) => void;
};

const MobileNav = ({ isOpen, isVisible, handleItemClick }: MobileNavProps) => {
  if (!isVisible) return null;

  return (
    <Container>
      <TransparentContainer onClick={handleItemClick}></TransparentContainer>
      <StyledMobileNav $isOpen={isOpen} data-testid='mobile-nav'>
        <NavItems />
      </StyledMobileNav>
    </Container>
  );
};

export default MobileNav;

const Container = styled.div`
  display: flex;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`;

const TransparentContainer = styled.div`
  flex-grow: 1;
  height: 100%;
  background-color: transparent;
`;

const StyledMobileNav = styled.nav<{ $isOpen: boolean }>`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  width: 100%;
  position: fixed;
  background-color: ${({ theme }) => theme.colors.background};
  height: 100%;
  padding-top: 5rem;
  animation: ${({ $isOpen }) => ($isOpen ? slideIn : slideOut)} 0.3s ease-in-out
    forwards;
  z-index: 5;
`;
