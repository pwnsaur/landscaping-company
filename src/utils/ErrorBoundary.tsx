import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
