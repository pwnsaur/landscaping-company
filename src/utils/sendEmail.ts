import axios, { AxiosResponse } from 'axios';

import { FormData } from '@/types/contactForm';

type ServerResponse = {
  messageId: string;
  response: string;
};

type SendEmailFn = (
  formData: FormData
) => Promise<AxiosResponse<ServerResponse>>;

const sendEmail: SendEmailFn = async (formData: FormData) => {
  return axios({
    method: 'post',
    url: '/api/send-mail',
    data: formData,
  });
};

export default sendEmail;
