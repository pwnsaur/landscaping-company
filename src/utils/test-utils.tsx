import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
