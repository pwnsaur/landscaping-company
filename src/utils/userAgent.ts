export function isMobileUserAgent(userAgent: string): boolean {
  return /Android|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
}
