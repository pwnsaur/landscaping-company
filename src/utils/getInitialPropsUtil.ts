import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
