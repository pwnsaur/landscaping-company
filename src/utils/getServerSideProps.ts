import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
