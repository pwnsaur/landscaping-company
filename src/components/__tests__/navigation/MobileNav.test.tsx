import MobileNav from '@/components/navigation/MobileNav';
import { render } from '@/utils/test-utils';

describe('MobileNav component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(
      <MobileNav isOpen={false} isVisible={true} handleItemClick={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
