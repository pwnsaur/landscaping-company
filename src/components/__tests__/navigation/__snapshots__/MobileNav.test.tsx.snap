// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNav component matches the snapshot 1`] = `
<DocumentFragment>
  .c3 {
  display: flex;
  align-items: center;
  font-size: 1.2rem;
  color: black;
  text-transform: uppercase;
  height: 4rem;
  padding: 0 1rem;
  white-space: nowrap;
  transition: color 0.3s ease-in-out;
}

.c3:hover {
  color: #696767;
  text-decoration: none;
}

.c0 {
  display: flex;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c1 {
  flex-grow: 1;
  height: 100%;
  background-color: transparent;
}

.c2 {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  width: 70%;
  position: fixed;
  background-color: #228b22;
  height: 100%;
  padding-top: 5rem;
  animation: kIDQyd 0.1s ease-in-out forwards;
  z-index: 5;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <nav
      class="c2"
      data-testid="mobile-nav"
    >
      <a
        class="c3"
        href="/"
      >
        SƒÅkums
      </a>
      <a
        class="c3"
        href="/services"
      >
        Pakalpojumi
      </a>
      <a
        class="c3"
        href="/projects"
      >
        Projekti
      </a>
      <a
        class="c3"
        href="/about"
      >
        Par mums
      </a>
      <a
        class="c3"
        href="/contacts"
      >
        Kontakti
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
